src/activity-logger/background/index.js

```diff
@@ -18,23 +21,57 @@ function logPageVisit({url}) {
     return storeVisit({timestamp, visitInfo, pageInfo})
 }
 
-// Listen for navigations to log them and analyse the pages.
-browser.webNavigation.onCommitted.addListener(details => {
+let pageLoaded = function(item){
+    pages.forEach((page)=>{
+        if(item === page){
```

Comment:
This tests strict object equality: given two objects whose properties are the same, it would still return false, unless they are really both references to the same object. Changing `===` to `==` will not change this either, you have to check for the equality of their properties (`tabId` and `url`)